docs/design/gitattributes-eol.md

```diff
@@ -0,0 +1,838 @@
+  * Unspecified. How EOL conversion should be applied is decided by the `working-copy.eol-conversion` setting and the `working-copy.gitattributes-default-eol` setting.
+  * All other cases. Act as if `eol` is in the unspecified state.
+  * Specifying `eol` automatically sets `text` if `text` was left unspecified.
+* Support the [`crlf` `gitattributes`][git-attr-crlf-compat]:
+  * Set. The same as `text` is set.
+  * Unset. The same as `text` is unset.
+  * Set to string value `input`. The same as `eol` is set to `lf`.
+* How the [`GitAttributes`] type introduced in the [basic `gitattributes` design](gitattributes.md) should be initialized and used.
+* If no EOL should be applied to the file, the file should not be read when calling the EOL convert functions(`convert_eol_for_snapshot` and `convert_eol_for_update`).
+
+[git-config-core-eol]: https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreeol
+[git-attr-crlf-compat]: https://git-scm.com/docs/gitattributes#_backwards_compatibility_with_crlf_attribute
+
+## State of the Feature as of v0.35.0
+
+* We have [a design doc](gitattributes.md) on how to read the `gitattributes` associated with a file.
+* We have implemented a feature similar to the git `core.autocrlf` config in [this PR][jj-pr-6728], which implements the EOL conversion itself.
+
+[jj-pr-6728]: https://github.com/jj-vcs/jj/pull/6728
+
+## Prior work
+
+* In git, it's [`crlf_to_git`][git-code-crlf-to-git] and [`crlf_to_worktree`][git-code-crlf-to-worktree].
+* In gitoxide, it's [`gix_filter::eol::convert_to_git`][gix-eol-convert-to-git] and [`gix_filter::eol::convert_to_worktree::convert_to_worktree`][gix-eol-convert-to-worktree].
+* In fig/mercurial, the [`hgeol`][hg-eol-extension] extension is used. Unlike git, where a repo can have multiple `gitattributes` files in different subdirectory, `hg` only allows one single `.hgeol` file at the root of the repo. [This][monetdb-hgeol-example] is an example of the contents. `fig`'s implementation have severe issues on Windows, where if both 2 changes modify a file, and you rebase one change on top of another, causing a merge, `fig` can't handle the EOL conversion correctly, and always introduces unexpected EOL changes.
+
+[git-code-crlf-to-git]: https://github.com/git/git/blob/9a2fb147f2c61d0cab52c883e7e26f5b7948e3ed/convert.c#L503-L590
+[git-code-crlf-to-worktree]: https://github.com/git/git/blob/9a2fb147f2c61d0cab52c883e7e26f5b7948e3ed/convert.c#L592-L627
+[gix-eol-convert-to-git]: https://docs.rs/gix-filter/0.21.0/gix_filter/eol/fn.convert_to_git.html
+[gix-eol-convert-to-worktree]: https://docs.rs/gix-filter/0.21.0/gix_filter/eol/convert_to_worktree/fn.convert_to_worktree.html
+[hg-eol-extension]: https://wiki.mercurial-scm.org/EolExtension
+[monetdb-hgeol-example]: https://github.com/MonetDB/MonetDB/blob/2bded273ab192a022e967bc2fa4db41bbc369ca9/.hgeol
+
+## Open questions
+
+1. Do we need more detailed description in the [Prior work](#prior-work) section?
```

Comment:
FWIW, I don't think so.

Reply:
Removed the section as we have received comments from 3 reviewers, although 2 of them are Googlers.